plugins/rbac/src/api/RBACBackendClient.ts

```diff
@@ -162,5 +188,63 @@ export class RBACBackendClient implements RBACAPI {
       },
       body: JSON.stringify(role),
     });
+    if (jsonResponse.status !== 200 && jsonResponse.status !== 201) {
+      return jsonResponse.json();
+    }
+    return jsonResponse;
+  }
+
+  async updateRole(oldRole: Role, newRole: Role) {
+    const { token: idToken } = await this.identityApi.getCredentials();
+    const backendUrl = this.configApi.getString('backend.baseUrl');
+    const { kind, namespace, name } = getKindNamespaceName(oldRole.name);
+    const body = {
+      oldRole,
+      newRole,
+    };
+    const jsonResponse = await fetch(
+      `${backendUrl}/api/permission/roles/${kind}/${namespace}/${name}`,
+      {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+          ...(idToken && { Authorization: `Bearer ${idToken}` }),
+        },
+        body: JSON.stringify(body),
+      },
+    );
+    if (jsonResponse.status !== 200 && jsonResponse.status !== 201) {
```

Comment:
Cool, this is a variation on the code snippet I mentioned in another comment @debsmita1 and should work fine.